lib/analyzer/applications/java.ts

```diff
@@ -46,6 +40,7 @@ export async function jarFilesToScannedProjects(
 
   for (const path in mappedResult) {
     if (!mappedResult.hasOwnProperty(path)) {
+      // todo is there a case that this condition will be true?
```

Comment:
Generally speaking, I try to avoid adding `todo`s to the code. Instead, I write a jira ticket with the todo. This will usually help get it prioritised, and is easier to track.
In this specific case, if you don't have the capacity to try to answer the question, I'd consider adding a ticket to research/add a test for what happens if the condition is true.